Added code to opDispatch to make a full example that can be copy, pas…

MikeShah opened the pull request:
Added code to opDispatch to make a full example that can be copy, pasted and run. I believe this will more clearly help users understand opDispatch. For clarity, I changed the incoming type to spell out 'Type'

MikeShah committed:
Added code to opDispatch to make a full example that can be copy, pasted and run. I believe this will more clearly help users understand opDispatch. For clarity, I changed the incoming type to spell out 'Type'

ntrel commented on gems/opdispatch-opapply.md:
```diff
@@ -16,20 +16,25 @@ template parameter. `opDispatch` is a *catch-all*
+		void callA(int i, int j) { writeln("C.callA\n"); }
+		void callB(string s) 	 { writeln("C.callB\n"); }
+	}
+	struct CallLogger(Type) {
+		Type content;
+		void opDispatch(string name, T...)(T vals) {
```
`T` should mean type, so it's clearer to say:
```suggestion
	struct CallLogger(T) {
		T content;
		void opDispatch(string name, Types...)(Types vals) {
```

ntrel commented:
BTW There are other examples such as [here](https://tour.dlang.org/tour/en/basics/type-qualifiers) that don't declare `main`.

MikeShah committed:
Update gems/opdispatch-opapply.md

Agreed, initially I thought 'Type' with 'Types' was a bit clearer, but the convention is purely 'T'

Co-authored-by: Nick Treleaven <ntrel002@gmail.com>

MikeShah committed:
Update gems/opdispatch-opapply.md

Co-authored-by: Nick Treleaven <ntrel002@gmail.com>

MikeShah commented:
> BTW There are other examples such as [here](https://tour.dlang.org/tour/en/basics/type-qualifiers) that don't declare `main`.

My thoughts are for the call logger example it is something new enough enough that a beginner might want to copy & paste. :)

thewilsonator committed:
Merge branch 'master' into opDispatchUpdate

thewilsonator committed:
Merge branch 'master' into opDispatchUpdate
